app/webpack/javascripts/modules/selectAll_spec.mjs

```diff
@@ -0,0 +1,88 @@
+import { SelectAll } from './selectAll.mjs'
+
+describe('SelectAll', () => {
+  describe('toggleSelection', () => {
+    let selectAllBox = null
+    let selectedBox = null
+    let unselectedBox = null
+    let testArea = null
+
+    beforeEach(() => {
+      document.body.classList.add('govuk-frontend-supported')
+
+      testArea = document.createElement('div')
+      document.body.appendChild(testArea)
+
+      // <div data-module="govuk-select-all" data-select-all-class="selector-box" data collection-class="pick-me">
+      //   <input class="selector-box" type="checkbox" />
+      // </div>
+      const selectAllDiv = document.createElement('div')
+      selectAllDiv.setAttribute('data-module', 'govuk-select-all')
+      selectAllDiv.setAttribute('data-select-all-class', 'selector-box')
+      selectAllDiv.setAttribute('data-collection-class', 'pick-me')
+      selectAllBox = document.createElement('input')
+      selectAllBox.type = 'checkbox'
+      selectAllBox.classList.add('selector-box')
+      selectAllBox.checked = false
+      selectAllDiv.appendChild(selectAllBox)
+      testArea.appendChild(selectAllDiv)
+
+      // <input class="pick-me" type="checkbox" name="box1" checked=true>
+      selectedBox = document.createElement('input')
+      selectedBox.type = 'checkbox'
+      selectedBox.classList.add('pick-me')
+      selectedBox.name = 'box1'
+      selectedBox.checked = true
+      testArea.appendChild(selectedBox)
+
+      // <input class="pick-me" type="checkbox" name="box2">
+      unselectedBox = document.createElement('input')
+      unselectedBox.type = 'checkbox'
+      unselectedBox.classList.add('pick-me')
+      unselectedBox.name = 'box2'
+      unselectedBox.checked = false
+      testArea.appendChild(unselectedBox)
+
```

Comment:
For above codes, you can consider creating a function that creates checkbox.
```javascript
      const createCheckbox = (className, checked = false, name = '') => {
        const checkbox = document.createElement('input');
        checkbox.type = 'checkbox';
        checkbox.classList.add(className);
        checkbox.name = name;
        checkbox.checked = checked;
        testArea.appendChild(checkbox);
        return checkbox;
      };
```
Then calling it like this
```javascript
unselectedBox = createCheckbox('pick-me', false, 'box2');
```